app/integration/runtime-camelk/pom.xml

```diff
@@ -76,6 +76,12 @@
       <artifactId>camel-k-runtime-core</artifactId>
       <version>${camel-k-runtime.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.apache.camel.k</groupId>
+      <artifactId>camel-k-adapter-camel-2</artifactId>
+      <version>${camel-k-runtime.version}</version>
+      <scope>runtime</scope>
```

Comment:
this should be in scope compile and you should exclude it from checks like:
```     <plugin>
        <artifactId>maven-dependency-plugin</artifactId>
        <configuration>
          <ignoredUnusedDeclaredDependencies>
            <!-- false positive, doesn't compile without it -->
            <ignoredUnusedDeclaredDependency>org.apache.camel.k:camel-k-adapter-camel-2</ignoredUnusedDeclaredDependency>
          </ignoredUnusedDeclaredDependencies>
        </configuration>
      </plugin>```

Reply:
Having it in scope runtime works on my end, and `grep -r 'org.apache.camel.k.adapter'` doesn't give me anything in `src/main/java`.

@valdar why do you think it needs to be in compile scope?

Reply:
This bit from the [CI build](https://circleci.com/gh/syndesisio/syndesis/71486)?

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /workspace/app/integration/runtime-camelk/src/main/java/io/syndesis/integration/runtime/camelk/logging/IntegrationLoggingContextCustomizer.java:[44,24] cannot access org.apache.camel.k.adapter.Registry
  class file for org.apache.camel.k.adapter.Registry not found
[ERROR] /workspace/app/integration/runtime-camelk/src/main/java/io/syndesis/integration/runtime/camelk/logging/IntegrationLoggingContextCustomizer.java:[45,24] cannot find symbol
  symbol:   method bind(java.lang.String,io.syndesis.integration.runtime.logging.BodyLogger.Default)
  location: variable runtimeRegistry of type org.apache.camel.k.Runtime.Registry
[ERROR] /workspace/app/integration/runtime-camelk/src/main/java/io/syndesis/integration/runtime/camelk/logging/IntegrationLoggingContextCustomizer.java:[47,24] cannot find symbol
  symbol:   method bind(java.lang.String,io.syndesis.integration.runtime.logging.ActivityTrackingInterceptStrategy)
  location: variable runtimeRegistry of type org.apache.camel.k.Runtime.Registry
```

Hmm, odd.

Reply:
I think the issue is that the `org.apache.camel.k.adapter.Registry` version the `org.apache.camel.k.Runtime` is linked to is not available on classpath.

I don't think you should do stuff like this:

https://github.com/apache/camel-k-runtime/blob/3e9d8f4943a7f89328aec43f9d37cfc59de467cc/camel-k-runtime-core/pom.xml#L120-L153

In `camel-k`, perhaps a `Registry` should be an interface placed in `camel-k-runtime-core` and have two implementations in `camel-k-adapter-camel-2` and `camel-k-adapter-camel-3`. Would that cause circular dependencies?

Reply:
you have a good point there, we must investigate further; should we merge this and rise another issue? Since might take time if a change in camel-k is needed and need to be released. wdyt @zregvart ?

Reply:
Agree, we should put that stuff in a common layer.

Reply:
it is a little bit more complicated than adding a common layer as the registry interface has changed in camel 3, so either we create yet another registry interface or we just wait to love to camel 3

Reply:
it is a little bit more complicated than adding a common layer as the registry interface has changed in camel 3, so either we create yet another registry interface or we just wait to love to camel 3

Reply:
never wait to love :P